tests/features/kill_replica.feature

```diff
@@ -285,3 +285,66 @@ Feature: Destroy synchronous replica in various scenarios
     Examples: <lock_type>
         | lock_type | lock_host  |
         | zookeeper | zookeeper1 |
+
+    @detached_replica
+    Scenario Outline: Disconnecting replica pgboncer behaviour
+        Given a "pgconsul" container common config
+        """
+            pgconsul.conf:
+                global:
+                    priority: 0
+                    use_replication_slots: '<use_slots>'
+                    quorum_commit: '<quorum_commit>'
+                primary:
+                    change_replication_type: 'yes'
+                    primary_switch_checks: 1
+                replica:
+                    allow_potential_data_loss: 'no'
+                    primary_unavailability_timeout: 1
+                    primary_switch_checks: 1
+                    min_failover_timeout: 1
+                    primary_unavailability_timeout: 2
+                    close_detached_after: 30
+                commands:
+                    generate_recovery_conf: /usr/local/bin/gen_rec_conf_<with_slots>_slot.sh %m %p
+        """
+        Given a following cluster with "<lock_type>" <with_slots> replication slots
+        """
+            postgresql1:
+                role: primary
+            postgresql2:
+                role: replica
+        """
+        Then <lock_type> "<lock_host>" has holder "pgconsul_postgresql1_1.pgconsul_pgconsul_net" for lock "/pgconsul/postgresql/leader"
+        Then container "postgresql2" is in <replication_type> group
+        Then <lock_type> "<lock_host>" has following values for key "/pgconsul/postgresql/replics_info"
+        """
+          - client_hostname: pgconsul_postgresql2_1.pgconsul_pgconsul_net
+            state: streaming
+        """
+        When we disconnect from network container "postgresql2"
+        And we wait "5.0" seconds
+        Then pgbouncer is running in container "postgresql2"
+        When we wait "30.0" seconds
+        Then pgbouncer is not running in container "postgresql2"
+        When we connect to network container "postgresql2"
+        Then container "postgresql1" is primary
+        Then container "postgresql2" is a replica of container "postgresql1"
+        Then container "postgresql2" is in <replication_type> group
```

Comment:
Then pgbouncer is running in container "postgresql2" ?

Reply:
Right, it is not guaranteed that replica in sync is open, it should be tested.